packages/components/src/internal/app/actions.ts

```diff
@@ -57,7 +59,26 @@ export function menuInit(currentProductId: string, appProductId: string) {
 export const menuInvalidate = () => ({ type: MENU_INVALIDATE });
 
 // an alternative to menuInvalidate, which doesn't erase current menu during reload
-export const menuReload = () => ({ type: MENU_RELOAD });
+export function menuReload() {
+    return (dispatch, getState) => {
+        dispatch({ type: MENU_RELOAD });
+        const menu = getState().routing.menu;
+        menu.getMenuSections()
```

Comment:
I am a little confused as to why this is happening here instead of just having the websocket listener dispatch the MENU_RELOAD as it did before and then the reducer for that dispatch type be the one that does this call?

Reply:
Also, this looks to be almost a copy of the code from the menuInit function, should that be changed to call this now?

Reply:
Because reducers don't kick off requests for data, they only change the state. Things like data fetching happen outside of reducers, typically in a thunk, and then when they're done getting data they `dispatch` their changes.

This code is similar to `menuInit`, but different enough that I separated it, and since it's only used by our websocket I think we're fine.